gr-soapy/grc/soapy_sdrplay_source.block.yml

```diff
@@ -86,17 +105,44 @@ templates:
       self.${id}.set_sample_rate(0, ${samp_rate})
       self.${id}.set_bandwidth(0, ${bandwidth})
       self.${id}.set_antenna(0, ${antenna})
-      self.${id}.set_gain_mode(0, ${agc})
       self.${id}.set_frequency(0, ${center_freq})
       self.${id}.set_frequency_correction(0, ${freq_correction})
-      self.${id}.set_gain(0, min(max(-${gain}, 0.0), 47.0))
+      self.${id}.set_gain_mode(0, ${agc})
+      ##
+      ## Bounds check. Valid values for LNA state (RFGR here)
+      ## vary by model, so runtime changes may cause errors. Set
+      ## gain (IFGR) only if agc is off.
+      self.${id}.set_gain(0, 'RFGR', min(max(${lna_state}, 0), 9))
+      if not self.agc:
```

Comment:
Putting this source in a separate top block fails because it's trying to access `agc` that does not exist, in that top block.

I think this ought to be `if not ${agc}` -- however I'm unsure how the methods should access `self.gain` and `self.agc`

Reply:
"Putting this source in a separate top block" ... do you mean cutting/pasting the source code, or something you can do with GRC?

I'll change to `${agc}` to be more like other params. In the code I'm looking at, though, it doesn't seem like it makes much difference:
`self.agc = agc = True`

We're hitting the limit of what these templates can do!

Reply:
I do see that placing the block in a hier block does not work well.

Reply:
Now works fine in a hier block.